File: blind-75/arrays-and-hashing/two-sum/java/TwoSum.java
import java.util.HashMap;

public class TwoSum {
  public int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> diff = new HashMap<>();

    for (int i = 0; i < nums.length; i++) {
      if (diff.containsKey(target - nums[i])) {
        return new int[] {diff.get(target - nums[i]), i};
      }

      diff.put(nums[i], i);
    }

    return null;
  }
}

File: blind-75/arrays-and-hashing/two-sum/typescript/two-sum.ts
export function twoSum(nums: number[], target: number): number[] {
  const diff = new Map<number, number>();
  const ans: number[] = [];

  for (let i = 0; i < nums.length; i++) {
    if (diff.has(target - nums[i])) {
      ans.push(diff.get(target - nums[i])!, i);

      break;
    }

    diff.set(nums[i], i);
  }

  return ans;
}
